pyomo/contrib/pyros/uncertainty_sets.py

```diff
@@ -56,14 +67,14 @@ def __init__(self, **kwargs):
         """
         return
 
-    @property
+    @abc.abstractmethod
     def dim(self):
         """
         UncertaintySet dimension, e.g. dimension of uncertain parameters
         """
         raise NotImplementedError
 
-    @property
```

Comment:
I think you still need the @property decorator (because you are overriding this with a property), but I am not surewhich order they need to be in.

Reply:
I read that in python 3.3+, @property comes before @abc.abstractmethod. I'll fix this.